src/components/tx-flow/flows/ExecuteBatch/ReviewBatch.tsx

```diff
@@ -62,12 +63,17 @@ export const ReviewBatch = ({ params }: { params: ExecuteBatchFlowProps }) => {
     return getTxsWithDetails(params.txs, chain.chainId)
   }, [params.txs, chain?.chainId])
 
-  const multiSendContract = useMemo(() => {
+  const [multiSendContract] = useAsync(async () => {
     if (!chain?.chainId || !safe.version) return
-    return getReadOnlyMultiSendCallOnlyContract(chain.chainId, safe.version)
+    const contract = await getReadOnlyMultiSendCallOnlyContract(chain.chainId, safe.version)
+
+    if (contract) {
+      setMultisendContractAddress(await contract.getAddress())
```

Comment:
Why is setting state from useAsync a bad thing? Ah, because it is supposed to return a value and I'm kind of misusing it to both return a value, but also to update the state. Okay, will change it.